### ISSUE TITLE
Spawning a `PbrBundle` causes gizmo to disappear or flicker

### ISSUE THREAD

**Issue by Seldom-SE** [james-j-obrien/bevy_vector_shapes#42]
In this project (using `bevy_vector_shapes` 0.8)
```rust
use std::f32::consts::FRAC_PI_2;

use bevy::{
    color::palettes::css::{RED, WHITE},
    prelude::*,
};
use bevy_vector_shapes::{painter::ShapePainter, shapes::DiscPainter, ShapePlugin};

fn main() {
    App::new()
        .add_plugins((DefaultPlugins, ShapePlugin::default()))
        .add_systems(Startup, setup)
        .add_systems(Update, draw_gizmo)
        .run();
}

pub fn setup(
    mut cmds: Commands,
    mut meshes: ResMut<Assets<Mesh>>,
    mut mats: ResMut<Assets<StandardMaterial>>,
) {
    cmds.spawn(Camera3dBundle {
        transform: Transform::from_translation(Vec3::Y * 10.).looking_at(Vec3::ZERO, Dir3::Y),
        ..default()
    });
}

fn draw_gizmo(mut painter: ShapePainter) {
    painter.rotate_x(FRAC_PI_2);
    painter.hollow = true;
    painter.color = RED.into();
    painter.circle(2.);
}
```
the gizmo shows up correctly

![Screenshot_20240708_133756](https://github.com/james-j-obrien/bevy_vector_shapes/assets/38388947/b02d3388-ba32-4cae-bb58-10e6e79669ee)

but if I add a `PbrBundle`
```rust
cmds.spawn(PbrBundle {
    mesh: meshes.add(Cuboid::default()),
    material: mats.add(Color::from(WHITE)),
    ..default()
});
```
the gizmo disappears

![Screenshot_20240708_133706](https://github.com/james-j-obrien/bevy_vector_shapes/assets/38388947/45ad31c7-e8b7-4f5c-b2ff-810643fbe3ed)

or flickers

https://github.com/james-j-obrien/bevy_vector_shapes/assets/38388947/88338bec-3dfc-441a-a807-b76fd008193e

Here's a repro: https://github.com/BraymatterOrg/bloom_bug_repro/blob/843f674ce2a124ec8592953396e6bcc866eb25c5/src/main.rs
This did work in `bevy_vector_shapes` 0.7. We've reproed on Linux, Mac, and Windows.

**Comment by james-j-obrien** [comment]
Very interesting, there were a large amount of renderer changes as part of 0.14 so I'm sure it's related to those changes.

Will take a look when I have a chance, thanks for the report!

**Comment by Braymatter** [comment]
I wonder if this could be related to the new anisotropy fields/bindings added to the material? 

I remember having issues with conditionally compiled stuff in another project and bindings were conflicting on the (new at the time) tessalation fields.  

The fact that the gizmo stops working correctly after a StandardMaterial is added to the scene makes me think it's related to _something_ conditionally compiled

**Comment by james-j-obrien** [comment]
Interestingly your repro doesn't actually show an issue for me which explains how this got released broken.

I'm running windows on this GPU:
`AdapterInfo { name: "NVIDIA GeForce RTX 2080 SUPER", vendor: 4318, device: 7809, device_type: DiscreteGpu, driver: "NVIDIA", driver_info: "546.01", backend: Vulkan }`

That will make this a little tricky for me to diagnose. I'm not sure at the moment why usage of a `PbrBundle` would affect the rendering at all given that I'm using my own pipelines so my best shot right now is scanning the code for something common I am relying on.

**Comment by Braymatter** [comment]
Here is my machine's info which has the issue:

2024-07-10T20:19:10.349554Z  INFO bevy_render::renderer: AdapterInfo { name: "Apple M3 Max", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }

**Comment by Seldom-SE** [comment]
Here's my info on Linux: `AdapterInfo { name: "AMD Radeon RX 570 Series (RADV POLARIS10)", vendor: 4098, device: 26591, device_type: DiscreteGpu, driver: "radv", driver_info: "Mesa 24.1.3-arch1.1", backend: Vulkan }`
and on Windows: `AdapterInfo { name: "Radeon RX 570 Series", vendor: 4098, device: 26591, device_type: DiscreteGpu, driver: "AMD proprietary driver", driver_info: "22.20.27.09", backend: Vulkan }`

If there's anything I can do to help, lmk

**Comment by james-j-obrien** [comment]
> To be clear are you saying the provided repro is working as intended?

Yes, for me the shape shows up and displays properly, no flickering at all. You're probably right that it's some kind of binding conflict, but it's very hard to find blindly unfortunately.

**Comment by james-j-obrien** [comment]
Seems to be a bevy issue, the reason it wasn't repro-ing for me is it requries the `multi-threading` feature to cause non-deterministic system ordering, otherwise they run in an order that hides the bug.

**Comment by james-j-obrien** [comment]
Fixed in 0.14.1
